app/presenters/comment_presenter.rb

```diff
@@ -19,11 +20,25 @@ def as_api_response
       author:           PersonPresenter.new(author).as_api_json,
       created_at:       created_at,
       mentioned_people: build_mentioned_people_json,
-      reported:         current_user.present? && reports.where(user: current_user).exists?
+      reported:         current_user.present? && reports.exists?(user: current_user),
+      interactions:     build_interactions_json
+    }
+  end
+
+  def build_interactions_json
+    {
+      likes:       as_api(likes),
```

Comment:
Hm - we do need this information to generate "like" or "unlike" buttons. For this a fetch has to be done. The correct link text is then done by the frontend. 
I do not have any idea if-  and how to change that.

Reply:
OH GOD, really? ... but why does this require ALL likes already and why are ALL likes then refetched anyway if they are already there? This isn't directed at you, it's more questioning why things were done that way in the past. But also it's not so much a problem so far for posts, as there are loaded 15 together at most, but for comments it could load hundreds of comments for a big post, so it becomes more of a problem now.

For the API I would just do what is also done for the posts API, as there is a `own_interaction_state` with booleans.

For the frontend I don't know how hard it is to also use such boolean? If it's too difficult to change the frontend to use a boolean instead of the list, we could just have the `likes` only containing the own like (as for rendering the likes there is a separate request fetching all likes again anyway), but this would be a pretty bad workaround, so I'm not sure how much I like it. But as the frontend is only an internal API and we will eventually replace it with a new frontend, it might be fine?

(And sorry, as written in the Review already, I didn't really look at the the code, so I didn't know the likes in the initial response is used for that, I only saw that the likes were already included in the initial response and then they are fetched again when the are shown, so I assumed the initial likes weren't used for anything)